docs/tutorials/getting-started-with-csharp/microservices.md

```diff
@@ -0,0 +1,443 @@
+#Introduction
+
+This tutorial details the tasks necessary to build and deploy
+an ASP.NET Core microservice in a Docker container. During the course
+of this tutorial, you'll learn:
+
+* How to generate an ASP.NET Core application using Yeoman
+* How to create a development Docker environment
+* How to build a Docker image based on an existing image.
+* How to deploy your service into a Docker container.
+
+Along the way, you'll also see some C# language features:
+
+* How to convert C# objects into JSON payloads.
+* Build immutable Data Transfer Objects
+* Process incoming HTTP Requests and generate the HTTP Response
+* Work with nullable value types
+
+## Why Docker?
+
+Docker makes it easy to create standard machine images to
+host your services in a data center, or the public cloud. Docker
+enables you to configure the image, and replicate it as needed to
+scale the installation of your application.
+
+All the code in this tutorial will work in any .NET Core environment.
+The additional tasks for a Docker installation will work for an ASP.NET
+Core application. 
+
+# Prerequisites
+You’ll need to setup your machine to run .NET core. Instructions are
+[here](http://dotnet.github.io/getting-started/). You can run this
```

Comment:
links saying here are bad for SEO. Consider rewriting this to something like "See the [getting started](http://dotnet.github.io/getting-started/) page for instructions on how to install .NET Core in your machine."